template/onchainKit/hooks/useEnsName.ts

```diff
@@ -1,34 +1,27 @@
-import { useEffect, useState } from 'react';
 import { publicClient } from '../store/client';
-import type { Address, GetEnsNameReturnType } from 'viem';
+import { inMemoryStorageService } from '../store/storageServices';
+import { ActionResponse } from '../types';
+import { useOnchainActionWithCache } from './useOnchainActionWithCache';
+import type { Address } from 'viem';
 
-/**
- * Fetches the ENS name for a given address.
- */
-export const useEnsName = (address?: Address) => {
-  const [ensName, setEnsName] = useState<GetEnsNameReturnType | null>(null);
-
-  useEffect(() => {
-    if (!address) return;
 
-    const fetchENSName = async () => {
-      try {
-        return await publicClient.getEnsName({
-          address,
-        });
-      } catch (err) {
-        return null;
-      }
-    };
 
-    fetchENSName()
-      .then((name) => {
-        setEnsName(name);
-      })
-      .catch((err) => {
-        console.log('err', err);
-      });
-  }, [address]);
+const ensNameAction = (address?: Address) => async (): Promise<ActionResponse> => {
+  if(!address) return undefined;
+  try {
+    return await publicClient.getEnsName({
+      address,
+    });
+  } catch (err) {
+    return undefined;
+  }
+};
 
+/**
+ * Fetches the ENS name for a given address.
+ */
+export const useEnsName = (address: Address) => {
+  const ensActionKey =  address ?? '';
+  const ensName = useOnchainActionWithCache(ensNameAction(address), ensActionKey, inMemoryStorageService);
```

Comment:
What our thinking around `ensActionKey`? 

Like should this be a key like `ens-name-${address}`?